usaspending_api/awards/v2/views/idvs/awards.py

```diff
@@ -0,0 +1,141 @@
+        ac.period_of_performance_start_date,
+        ac.piid
+    from
+        parent_award pap
+        inner join parent_award pac on pac.parent_award_id = pap.award_id
+        inner join awards ac on ac.id = pac.award_id
+        inner join transaction_fpds tf on tf.transaction_id = ac.latest_transaction_id
+    where
+        pap.{award_id_column} = %s
+    order by
+        {sort_column} {sort_direction}, ac.id {sort_direction}
+    limit {limit} offset {offset}
+""")
+
+GET_CONTRACTS_SQL = SQL("""
+    select
+        ac.id                                      award_id,
+        ac.type_description                        award_type,
+        ac.description,
+        tf.funding_agency_name                     funding_agency,
+        ac.generated_unique_award_id,
+        tf.ordering_period_end_date                last_date_to_order,
+        ac.total_obligation                        obligated_amount,
+        ac.period_of_performance_current_end_date,
+        ac.period_of_performance_start_date,
+        ac.piid
+    from
+        parent_award pap
+        inner join awards ap on ap.id = pap.award_id
+        inner join awards ac on ac.fpds_parent_agency_id = ap.fpds_agency_id and ac.parent_award_piid = ap.piid and
+            ac.type not like 'IDV\_%%'
+        inner join transaction_fpds tf on tf.transaction_id = ac.latest_transaction_id
+    where
+        pap.{award_id_column} = %s
+    order by
+        {sort_column} {sort_direction}, ac.id {sort_direction}
+    limit {limit} offset {offset}
+""")
+
+
+def _prepare_tiny_shield_rules():
+    """
+    Our TinyShield rules never change.  Encapsulate them here and store them
+    once in TINY_SHIELD_RULES.
+    """
+    models = customize_pagination_with_sort_columns(SORTABLE_COLUMNS, DEFAULT_SORT_COLUMN)
+    models.extend([
+        get_internal_or_generated_award_id_rule(),
+        {'key': 'idv', 'name': 'idv', 'type': 'boolean', 'default': True, 'optional': True}
+    ])
+    return TinyShield(models)
+
+
+TINY_SHIELD_RULES = _prepare_tiny_shield_rules()
+
+
+class IDVAwardsViewSet(APIDocumentationView):
+    """Returns the direct children of an IDV.
+    endpoint_doc: /awards/idvs/awards.md
+    """
+
+    @staticmethod
+    def _parse_and_validate_request(request: Request) -> dict:
+        return TINY_SHIELD_RULES.block(request)
+
+    @staticmethod
+    def _business_logic(request_data: dict) -> list:
+        # By this point, our award_id has been validated and cleaned up by
+        # TinyShield.  We will either have an internal award id that is an
+        # integer or a generated award id that is a string.
+        award_id = request_data['award_id']
+        award_id_column = 'award_id' if type(award_id) is int else 'generated_unique_award_id'
+
+        sql = GET_IDVS_SQL if request_data['idv'] else GET_CONTRACTS_SQL
+        sql = sql.format(
+            award_id_column=Identifier(award_id_column),
+            award_id=Literal(award_id),
```

Comment:
Dumb question: since you are using Literal here, do you even need to pass `award_id` as a parameter on line 127?

If we want to keep the parameter, then is `award_id` here redundant?

Reply:
Man.  You're determined not to let anything through, are you!

This is another good catch.  Also probably an artifact from refactoring.  I vaguely remember experimenting to see if it mattered which way I passed in the award id.